solo-chains/docs/bridge/etbereum-to-solo-chain.md

```diff
@@ -0,0 +1,116 @@
+        assert is_valid_merkle_branch(
+            leaf=hash_tree_root(update.header),
+            branch=update.finality_branch,
+            depth=floorlog2(FINALIZED_ROOT_INDEX),
+            index=get_subtree_index(FINALIZED_ROOT_INDEX),
+            root=update.finality_header.state_root,
+        )
+
+    # Verify update next sync committee if the update period incremented
+    if update_period == snapshot_period:
+        sync_committee = snapshot.current_sync_committee
+        assert update.next_sync_committee_branch == [Bytes32() for _ in range(floorlog2(NEXT_SYNC_COMMITTEE_INDEX))]
+    else:
+        sync_committee = snapshot.next_sync_committee
+        assert is_valid_merkle_branch(
+            leaf=hash_tree_root(update.next_sync_committee),
+            branch=update.next_sync_committee_branch,
+            depth=floorlog2(NEXT_SYNC_COMMITTEE_INDEX),
+            index=get_subtree_index(NEXT_SYNC_COMMITTEE_INDEX),
+            root=update.header.state_root,
+        )
+
+    # Verify sync committee has sufficient participants
+    assert sum(update.sync_committee_bits) >= MIN_SYNC_COMMITTEE_PARTICIPANTS
+
+    # Verify sync committee aggregate signature
+    participant_pubkeys = [pubkey for (bit, pubkey) in zip(update.sync_committee_bits, sync_committee.pubkeys) if bit]
+    domain = compute_domain(DOMAIN_SYNC_COMMITTEE, update.fork_version, genesis_validators_root)
+    signing_root = compute_signing_root(signed_header, domain)
+    assert bls.FastAggregateVerify(participant_pubkeys, signing_root, update.sync_committee_signature)
+```
+
+## Transmiting Ethereum Headers to Starlight
+
+### The relayers job to transmit Ethereum's header
+1. The relayer node needs to collect the aggergated sync committee signature embeedded into the Ethereum block
+2. The relayer needs to aggregrate the public keys responsible for generating that block.
+3. If the sync committee changes, the relayer needs to generate a merkle proof proving that the sync committee has changed.
+4. The relayer presents all these components to the appropriate pallet in starlight
+
+### Eth state receiver pallets job
+1. If the relayer claims the sync committee changes, verify the merkle proof to make sure the new sync committee is correct.
+2. Perform a verification of the validity of the committee members that signed the aggregated message
+3. Perform a verification of the aggregated signature
+4. If all of those pass, accept the header, with the new **state root ready to receive storage proofs**
+
+### Overall Diagram ETH state validity Reception
+
+<p align="center">
+  <img src="images/beacon_chain_altair.drawio.png" width="1000">
+</p>
+
+
+
+## Proving symbiotic validator selection to starlight (or any other Ethereum state)
+
+### The relayers job to transmit symbiotic validators
+1. The relayer node checks the latest accepted header in starlight, and it generates a storage proof of the validators storage item in the symbiotic smart contract
```

Comment:
This will be something like, the gateway contract will emit an event, which will be verified against the transaction receipt trie root containing in the block.

Reply:
added this explanation